ethcore/src/state.rs

```diff
@@ -56,6 +56,22 @@ impl AccountEntry {
 			AccountEntry::Missing => false,
 		}
 	}
+
+	fn clone_dirty(&self) -> Option<AccountEntry> {
+		match *self {
+			AccountEntry::Cached(ref acc) if acc.is_dirty() => Some(AccountEntry::Cached(acc.clone_dirty())),
+			AccountEntry::Killed => Some(AccountEntry::Killed),
+			_ => None,
+		}
+	}
+
+	fn clone_basic(&self) -> AccountEntry {
```

Comment:
maybe add some docs for this function - difficult to immediately see what it's trying to do